Make.config

```diff
@@ -32,26 +32,26 @@ SOM_PACKAGE_VERSION=0.2.$(SOM_PACKAGE_VERSION_REV).$(SOM_COMMIT_DISTANCE)
 
 # The Xcode we need.
 # Use the path as used in the Azure Pipelines vm images.
-XCODE_VERSION=12.0
-XCODE_URL=http://xamarin-storage/bot-provisioning/xcodes/Xcode_12.0.0.xip
-XCODE_DEVELOPER_ROOT=/Applications/Xcode_12.0.0.app/Contents/Developer
+XCODE_VERSION=12.0.1
+XCODE_URL=http://xamarin-storage/bot-provisioning/xcodes/Xcode_12.0.1.xip
+XCODE_DEVELOPER_ROOT=/Applications/Xcode_12.0.1.app/Contents/Developer
 
 # Minimum Mono version we need
-MIN_MONO_VERSION=6.4.0.0
-MAX_MONO_VERSION=6.4.0.999
-MIN_MONO_URL=https://download.visualstudio.microsoft.com/download/pr/5b7dcb51-3035-46f7-a8cb-efe3a1da351c/dcba976cd3257636b6b2828575d29d3c/monoframework-mdk-6.4.0.208.macos10.xamarin.universal.pkg
+MIN_MONO_VERSION=6.12.0.90
+MAX_MONO_VERSION=6.12.0.90
+MIN_MONO_URL=https://download.mono-project.com/archive/6.12.0/macos-10-universal/MonoFramework-MDK-6.12.0.90.macos10.xamarin.universal.pkg
 
 # XI version we need
-MIN_XI_VERSION=12.0.0.15
-MAX_XI_VERSION=12.0.0.15
-MIN_XI_URL=https://download.visualstudio.microsoft.com/download/pr/28396824-ecde-429e-9693-0da9382e1474/debd3c099bf5a516b964675c2628e601/xamarin.ios-12.0.0.15.pkg
+MIN_XI_VERSION=13.18.2.1
+MAX_XI_VERSION=13.18.2.1
```

Comment:
You might run off into some errors (like `mlauch`) since 13.18 was released before Xcode 12.
What's blocking you from using XI 14.x ?

Reply:
When using XI 14.x, I get errors like this "ld: warning: object file (bin/Debug/mac/x86_64/ofiles/registeraccess.o) was built for newer **macOS** version (10.15) than being linked (10.9)".

I was having trouble tracking down this error. Steve told me what versions he was using and succeeding with so I tried his versions (which seemed compatible with macOS 10.15) so I transitioned to his XI 13.18.2.1. 
However, after you mentioned to bump MIN_MACOS_BUILD_VERSION to 10.15.4, that error up there changes to "ld: warning: object file (bin/Debug/appletv/arm64/ofiles/registeraccess.o) was built for newer **tvOS** version (14.0) than being linked (10.2)" 

Now I feel like there is just a tvOS version being set somewhere, but I have not seen it yet.

Reply:
You should be able to run `make V=1` in swiftglue/ and get a more verbose log (and one where everything is run serialized, so the warning will apply to the last command shown).

In any case, this is probably because of a missing compiler flag somewhere (my guess would be `-mmacosx-version-min=10.9`, and the equivalent for the other platforms), but I might be wrong here.

Reply:
Oops, my apologies. I was looking at the warning and not the error! The error has to do with modules not being found in this section "SWIFTC [iphone/x86_64/Release]  XamGlue" at https://dev.azure.com/xamarin/public/_build/results?buildId=30396&view=logs&j=b944efd9-1a3a-54bf-926f-a01c570708bc&t=21e7d97e-20fa-52e2-1c14-34bcaa2b17dd&l=259

This happens regardless of the minimum set macOS version and changing the XI from 13.18.2.1 to 14.2.0.12 seems to trigger this. This happens to me locally as well.

Reply:
Wild guess: you might have to add `MediaSetup` to the list of frameworks here: https://github.com/xamarin/binding-tools-for-swift/blob/e54898827eaf58c9d3e4c8b0c90fbe27e65f5a81/common.mk#L137

Reply:
Oh man.. Thank you haha I'll give that a try!

Reply:
Sadly this did not fix this issue. Perhaps I will mark this as an issue and attempt to fix this in a later PR since there are other fixes in this PR that are blocking @chamons. Is this okay?